src/main/java/com/gorlah/kappabot/rest/model/CommandRequest.java

```diff
@@ -1,11 +1,22 @@
 package com.gorlah.kappabot.rest.model;
 
+import com.gorlah.kappabot.function.BotRequestEndpoint;
 import com.gorlah.kappabot.function.BotRequestMetadata;
 import lombok.Value;
 
 @Value
 public class CommandRequest implements BotRequestMetadata {
```

Comment:
Still feels a bit weird to me that each of these is an implementation of the interface. I think rather than having an implementation for Discord and an implementation for REST, it may be cleaner to make `BotRequestMetadata` a class or at least provide something like a `SimpleBotRequestMetadata` implementation (might be a good use of `@Builder`). I can explain this more if needed.

Reply:
This is handled by changes in master.